packages/core/src/base-axis-chart.ts

```diff
@@ -136,6 +142,107 @@ export class BaseAxisChart extends BaseChart {
 		}));
 	}
 
+	createSlider(){
+
+		//TODO get the name of the action bar for the chart
+		var divToInsert = "#actions-" + this.options.type
+		console.log(divToInsert)
+
+		var handle_radius = 50,
+			handle_start_val = 50,
+			handle_padding = 10,
+			slider_height = 5,
+			axis_height = 20;
+
+		//insert the slider in the action bar element
+		var setup = marcon()
+				.width(window.innerWidth)
+				.height(handle_radius * 2 + slider_height + handle_padding + axis_height)
+				.left(handle_radius)
+				.right(handle_radius)
+				.bottom(axis_height)
+				.element(divToInsert);
+
+		setup.render();
+
+		var range_width = setup.innerWidth(), range_height = setup.innerHeight(), range_svg = setup.svg();
+
+		var range_x = scaleLinear()
+				.range([0, range_width])
+				.domain([0, 100]);
+
+		// the range axis
+		range_svg.append("g")
+				.attr("class", "range-axis")
+				.attr("transform", "translate(0, " + range_height + ")")
+				.call(axisBottom(range_x))
+
+		// the range scale
+		range_svg.append("rect")
+				.attr("class", "range-body")
+				.attr("x", 0)
+				.attr("y", range_height - slider_height)
+				.attr("width", range_width)
+				.attr("height", slider_height);
+
+		// the handle
+		range_svg.append("circle")
+				.attr("class", "range-dragger range-handle")
+				.attr("cx", range_x(handle_start_val))
+				.attr("cy", range_height - slider_height - handle_radius - handle_padding)
+				.attr("r", handle_radius)
+				.call(drag()
+					.on("drag", dragged)
+				);
+
+		// the label
+		range_svg.append("text")
+				.attr("class", "range-label")
+				.attr("x", range_x(handle_start_val))
+				.attr("y", range_height - slider_height - handle_radius - handle_padding)
+				.attr("dy", ".3em")
+				.text(handle_start_val);
+
+		// the pointer
+		range_svg.append("polygon")
+				.attr("class", "range-dragger range-pointer")
+				.attr("points", calcPointerPoints(handle_start_val))
+				.call(drag()
+					.on("drag", dragged)
+				);
+
+		function calcPointerPoints(handle_val){
+			var point_c = range_x(handle_val) + "," + (range_height - slider_height);
+			var point_a = (range_x(handle_val) - (handle_radius / 4)) + "," + (range_height - slider_height - handle_padding - (handle_radius / 10));
+			var point_b = (range_x(handle_val) + (handle_radius / 4)) + "," + (range_height - slider_height - handle_padding - (handle_radius / 10));
+			return point_a + " " + point_b + " " + point_c;
+		}
+
+		function dragged(){
```

Comment:
```js
const myFunc = (myVar) => {
   ...
};
```

Reply:
Won't let me use the 'this' reference ( coordinates = mouse(this) ).

Message: Argument of type 'this' is not assignable to parameter of type 'ContainerElement